File: Python/program/Car.py
from Vehicle import Vehicle

class Car(Vehicle):
    
    __jmlKursi = 0
    __jmlPintu = 0

    def __init__(self, platNomor="", merk="", tahunProduksi=-1, warna="", jmlKursi=0, jmlPintu=0):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jmlKursi = jmlKursi
        self.__jmlPintu = jmlPintu

    def getJmlKursi(self):
        return self.__jmlKursi

    def setJmlKursi(self, jmlKursi):
        self.__jmlKursi = jmlKursi

    def getJmlPintu(self):
        return self.__jmlPintu

    def setJmlPintu(self, jmlPintu):
        self.__jmlPintu = jmlPintu

    def getByColName(self, colName=""):
        if colName == "platNomor":
            return self.getPlatNomor()
        elif colName == "merk":
            return self.getMerk()
        elif colName == "tahunProduksi":
            return self.getTahunProduksi()
        elif colName == "warna":
            return self.getWarna()
        elif colName == "jmlKursi":
            return self.getJmlKursi()
        elif colName == "jmlPintu":
            return self.getJmlPintu()

File: CPP/Car.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle{
    private:
        int jmlKursi, jmlPintu;
    public:
        Car(){

        }
        Car(string platNomor, string merk, int tahunProduksi, string warna, int jmlKursi) : Vehicle(platNomor, merk, tahunProduksi, warna){
            this->jmlKursi = jmlKursi;
            this->jmlPintu = -1;
        }
        Car(string platNomor, string merk, int tahunProduksi, string warna, int jmlKursi, int jmlPintu) : Vehicle(platNomor, merk, tahunProduksi, warna){
            this->jmlKursi = jmlKursi;
            this->jmlPintu = jmlPintu;
        }

        void setJmlKursi(int jmlKursi){
            this->jmlKursi = jmlKursi;
        }
        void setJmlPintu(int jmlPintu){
            this->jmlPintu = jmlPintu;
        }

        int getJmlKursi(){
            return this->jmlKursi;
        }
        int getJmlPintu(){
            return this->jmlPintu ;
        }

        string getByColName(string colName){
            if(colName == "platNomor"){
                return this->getPlatNomor();
            }
            else if(colName == "merk"){
                return this->getMerk();
            }
            else if(colName == "warna"){
                return this->getWarna();
            }
            else if(colName == "tahunProduksi"){
                return to_string(this->getTahunProduksi());
            }
            else if(colName == "jmlKursi"){
                return to_string(this->getJmlKursi());
            }
            else if(colName == "jmlPintu"){
                return to_string(this->getJmlPintu());
            }
        }

        ~Car(){
            
        }
};